src/communication/xcConnection.ts

```diff
@@ -31,12 +31,14 @@ Connection.prototype.init = function (xcApiFileName, serverUrl, sessionData, ses
     let session = SessionFactory.create(serverUrl, null, sessionData);
     let getXcApiRequest = function (xcApiFileName, sessionListener) {
         session.privateSubscriber.getXcApi(xcApiFileName, function (xcApi) {
-            let parser = new Parser();
-            let configuration = new Configuration(parser);
-            configuration.init(xcApi);
-            session.configuration = configuration;
-            session.replyPublisher.configuration = configuration;
-            sessionListener(null, session);
+            const parser = new DefaultApiConfigurationParser();
+            const configurationPromise = parser.parse(xcApi);
+            configurationPromise.then(configuration => {
+                session.configuration = configuration;
+                session.replyPublisher.configuration = configuration;
+                sessionListener(null, session);
+            })
+            .catch(e => sessionListener(e, null));
         });
     };
     session.init(sessionListener, getXcApiRequest, xcApiFileName);
```

Comment:
Don't we risk initializing the session while the configuration is still parsing??